lib/jsanalyze.js

```diff
@@ -171,7 +172,37 @@ exports.analyzeJs = function analyzeJs(contents, opts) {
 	// transpile
 	if (opts.transpile) {
 		options.plugins.push(require.resolve('./global-this'));
-		options.presets.push([ env, { targets: opts.targets } ]);
+		options.plugins.push(require.resolve('babel-plugin-transform-async-to-generator'));
+		options.presets.push([ env, { targets: opts.targets, useBuiltIns: true } ]);
+
+		// install polyfill
+		if (opts.resourcesDir) {
+			const modulesDir = path.join(opts.resourcesDir, 'node_modules');
+
+			// make sure our 'node_modules' directory exists
+			if (!fs.existsSync(modulesDir)) {
+				fs.mkdirSync(modulesDir);
+			}
+
+			// copy over polyfill and its dependencies
+			[ 'babel-polyfill', 'core-js', 'regenerator-runtime' ].forEach((moduleName) => {
```

Comment:
So obviously this is much simpler than using npm install or that npm package I pointed at before about copying dependencies over. And it looks like for now this is comprehensive - but it does scare me for future maintenance that we're essentially hard-coding the known dependencies in an array here and not doing a more general solution that would adapt if the babel-polyfill dependencies changed in future releases.

I think at the very least we may want to leave a nice comment/scary notice here that that's exactly what we're doing and to be aware if updating babel-polyfill to look at/modify this logic.